content-security-policy/support/testharness-helper.sub.js

```diff
@@ -39,3 +55,53 @@ function assert_embedding_csp(t, url, csp, expected) {
 
   document.body.appendChild(i);
 }
+
+function assert_iframe_with_csp(t, url, csp, shouldBlock, urlId, blockedURI) {
+  var i = document.createElement('iframe');
+  i.src = url + "&id=" + urlId;
+  i.csp = csp;
+
+  var loaded = {};
+  window.addEventListener("message", function (e) {
+    if (e.source != i.contentWindow)
+        return;
+    if (e.data["loaded"])
+        loaded[e.data["id"]] = true;
+  });
+
+  if (shouldBlock) {
+    // Assert iframe does not load and is inaccessible.
+    window.onmessage = function (e) {
+      if (e.source != i.contentWindow)
+          return;
+      t.unreached_func('No message should be sent from the frame.');
+    }
+    i.onload = t.step_func(function () {
+      // Delay the check until the postMessage has a chance to execute.
+      setTimeout(t.step_func_done(function () {
+        assert_equals(loaded[urlId], undefined);
+      }), 1);
+
+      assert_throws("SecurityError", () => {
+        var x = i.contentWindow.location.href;
+      });
+    });
+  } else if (blockedURI) {
+    // Assert iframe loads with an expected violation.
+    window.addEventListener('message', t.step_func(e => {
+      if (e.source != i.contentWindow || !e.data["blockedURI"])
```

Comment:
Is it ever acceptable for a message from a frame which expects |blockedURI| to send a message that doesn't contain the value? Shouldn't we consider that a failure?